contrib/admin/mypy-with-ignore.py

```diff
@@ -91,6 +91,7 @@ def main():
         'src/toil/lib/encryption/conftest.py',
         'src/toil/lib/encryption/__init__.py',
         'src/toil/lib/aws/__init__.py',
+        'src/toil/lib/aws/iam.py',
```

Comment:
I still don't want to merge this with this in here. When I take it out, here's what I get from `make mypy`:

```
/home/anovak/workspace/toil/contrib/admin/mypy-with-ignore.py
src/toil/lib/aws/iam.py: note: In function "policy_permissions_allow":
src/toil/lib/aws/iam.py:54:116: error: Incompatible default for argument
"given_permissions" (default has type "Dict[<nothing>, <nothing>]", argument has
type "Set[str]")  [assignment]
    ...tr, List[str]] = {'*': []}, given_permissions: Set[str] = {}) -> bool:
                                                                 ^
src/toil/lib/aws/iam.py: note: In function "allowed_actions_roles":
src/toil/lib/aws/iam.py:103:50: error: Argument 1 to "dict" has incompatible
type "GetRolePolicyResponseTypeDef"; expected
"SupportsKeysAndGetItem[str, List[str]]"  [arg-type]
            role_policy: Dict[str, List[str]] = dict(iam.get_role_policy(
                                                     ^
src/toil/lib/aws/iam.py:103:50: note: Following member(s) of "GetRolePolicyResponseTypeDef" have conflicts:
src/toil/lib/aws/iam.py:103:50: note:     Expected:
src/toil/lib/aws/iam.py:103:50: note:         def __getitem__(self, str) -> List[str]
src/toil/lib/aws/iam.py:103:50: note:     Got:
src/toil/lib/aws/iam.py:103:50: note:         def __getitem__(self, str) -> object
src/toil/lib/aws/iam.py:108:26: error: No overload variant of "__getitem__" of
"list" matches argument type "str"  [call-overload]
            for statement in role_policy["PolicyDocument"]["Statement"]:
                             ^
src/toil/lib/aws/iam.py:108:26: note: Possible overload variants:
src/toil/lib/aws/iam.py:108:26: note:     def __getitem__(self, SupportsIndex) -> str
src/toil/lib/aws/iam.py:108:26: note:     def __getitem__(self, slice) -> List[str]
src/toil/lib/aws/iam.py:113:54: error: Argument 1 to "append" of "list" has
incompatible type "List[str]"; expected "str"  [arg-type]
    ...                allowed_actions[resource].append(role_policy[statement...
                                                        ^
src/toil/lib/aws/iam.py: note: In function "get_aws_account_num":
src/toil/lib/aws/iam.py:178:12: error: Incompatible return value type (got
"Optional[str]", expected "str")  [return-value]
        return boto3.client('sts').get_caller_identity().get('Account')
               ^
Found 5 errors in 1 file (checked 62 source files)
make: *** [Makefile:250: mypy] Error 1
```

I think we can probably sort these out with around half an hour of work.